os-eap-launch/added/launch/security-domains.sh

```diff
@@ -5,17 +5,65 @@ function prepareEnv() {
   unset SECDOMAIN_ROLES_PROPERTIES
   unset SECDOMAIN_LOGIN_MODULE
   unset SECDOMAIN_PASSWORD_STACKING
+
+  for prefix in $(echo $SECURITY_DOMAINS | sed "s/,/ /g"); do
+    clearDomainEnv $prefix
+  done
+  unset SECURITY_DOMAINS
+
+}
+
+function clearDomainEnv() {
+  local prefix=$1
+
+  unset ${prefix}_LOGIN_MODULE_CODE
+  unset ${prefix}_LOGIN_MODULE_MODULE
+
+  for option in $(compgen -v | grep -s "${prefix}_MODULE_OPTION_"); do
+    unset ${option}
+  done
 }
 
 function configure() {
+  configure_legacy_security_domains
   configure_security_domains
 }
 
 function configureEnv() {
-  configure
+  configure_security_domains
+}
+
+function configure_security_domains() {
+  if [ -n "$SECURITY_DOMAINS" ]; then
+    for domain_prefix in $(echo $SECURITY_DOMAINS | sed "s/,/ /g"); do
+      local security_domain="<security-domain name=\"$domain_prefix\" cache-type=\"default\">"
```

Comment:
Would it be better to explicitly specify a NAME suffix for the domain name?  This type of thing caused problems with the DB prefix mapping stuff (where the prefix had to be of the form <service-name>_<db-type>).